### PR TITLE
Add keyboard GRS-70EC

### PR DESCRIPTION
**Pull request by sekigon-gonnoc**
## Description

Add a new keyboard: GRS-70EC

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/sekigon/grs_70ec/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+             KC_TAB,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_HOME,  KC_PGDN, KC_PGUP, KC_END, KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,
+             KC_LCTL, KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_LEFT,  KC_DOWN, KC_UP,   KC_RGHT,KC_SCLN, KC_QUOT, KC_ENT,  LALT(KC_GRV),
+    KC_LSFT, KC_LSFT, KC_F11,  KC_F12,  KC_C,    KC_V,    KC_B,    KC_N,     KC_M,    KC_COMM, KC_DOT, KC_SLSH, KC_RSFT, KC_UP,   KC_RSFT,
+             KC_LCTL, KC_LGUI, KC_LALT, KC_SPC,  KC_SPC,  KC_TRNS,  KC_ENT,   KC_RGUI, KC_RALT,                 KC_LEFT, KC_DOWN, KC_RGHT
+    )
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    return true;
+}
```
Comment (zvecr) [inline keyboards/sekigon/grs_70ec/keymaps/default/keymap.c:50]:
```suggestion
```

File: keyboards/sekigon/grs_70ec/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+
+// Defines the keycodes used by our macros in process_record_user
+// enum custom_keycodes {
+//     QMKBEST = SAFE_RANGE,
+// };
+
```
Comment (zvecr) [inline keyboards/sekigon/grs_70ec/keymaps/default/keymap.c:23]:
```suggestion

```

File: keyboards/sekigon/grs_70ec/grs_70ec.c [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    writePinHigh(D2);
+}
+
+void led_off(void) { writePinLow(D2); }
+
+void keyboard_post_init_kb() {
+    debug_enable = true;
+    debug_matrix = true;
+
+    led_on();
```
Comment (zvecr) [inline keyboards/sekigon/grs_70ec/grs_70ec.c:30]:
```suggestion
    led_on();
    
    keyboard_post_init_user();
```

File: keyboards/sekigon/grs_70ec/grs_70ec.c [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+void keyboard_pre_init_kb(void) {
+    // Turn on extern circuit
+    setPinOutput(F7);
+    writePinHigh(F7);
```
Comment (zvecr) [inline keyboards/sekigon/grs_70ec/grs_70ec.c:36]:
```suggestion
    writePinHigh(F7);
    
    keyboard_pre_init_user();
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/sekigon/grs_70ec/rules.mk [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
```
Comment (fauxpark) [inline keyboards/sekigon/grs_70ec/rules.mk:27]:
```suggestion
# MCU name
MCU = atmega32u4

# Bootloader selection
BOOTLOADER = caterina

# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = yes        # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
CUSTOM_MATRIX = lite
SPLIT_KEYBOARD = yes

SRC += analog.c ec_switch_matrix.c matrix.c
```

File: keyboards/sekigon/grs_70ec/readme.md [hunk 6]
```diff
@@ -0,0 +1,19 @@
+# GRS-70EC
+
+![grs_70ec](https://raw.githubusercontent.com/sekigon-gonnoc/GRS-70EC-doc/master/img/GRS-70EC.JPG)
+
+* A split keyboard using electrostatic capacitive switch
```
Comment (fauxpark) [inline keyboards/sekigon/grs_70ec/readme.md:5]:
```suggestion
A split keyboard using electrostatic capacitive switch
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by mtei — APPROVED** [review]
Looks good to me.

**Comment by drashna** [comment]
Thanks!

File: keyboards/sekigon/grs_70ec/config.h [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+ *
+ * Change this to how you wired your keyboard
+ * COLS: AVR pins used for columns, left to right
+ * ROWS: AVR pins used for rows, top to bottom
+ * DIODE_DIRECTION: COL2ROW = COL = Anode (+), ROW = Cathode (-, marked on diode)
+ *                  ROW2COL = ROW = Anode (+), COL = Cathode (-, marked on diode)
+ *
+ */
+#define MATRIX_ROW_PINS { C6, D7, E6, B4, B5 }
+#define MATRIX_COL_PINS { S2, S1, S0, S3, S5, S7, S6, S4 }
```
Comment (fauxpark) [inline keyboards/sekigon/grs_70ec/config.h:48]: The API really does not seem to like this.
https://github.com/qmk/qmk_firmware/runs/3120879453

**Pull request merged**